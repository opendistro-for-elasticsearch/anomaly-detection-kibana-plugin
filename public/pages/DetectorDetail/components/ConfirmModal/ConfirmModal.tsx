/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiModal,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiModalBody,
  EuiSpacer,
  EuiText,
  EuiButtonEmpty,
  EuiButton,
  ButtonColor,
} from '@elastic/eui';

interface ConfirmModalProps {
  title: string;
  description: string | React.ReactNode;
  callout?: any;
  confirmButtonText: string;
  confirmButtonColor: ButtonColor;
  confirmButtonDisabled?: boolean;
  onClose(): void;
  onCancel(): void;
  onConfirm(): void;
}

export const ConfirmModal = (props: ConfirmModalProps) => {
  return (
    <EuiModal onClose={props.onClose}>
      <EuiModalHeader>
        <EuiModalHeaderTitle>{props.title}&nbsp;</EuiModalHeaderTitle>
      </EuiModalHeader>
      <EuiModalBody>
        <EuiFlexGroup direction="column">
          {props.callout ? (
            <EuiFlexItem grow={false}>{props.callout}</EuiFlexItem>
          ) : null}
          <EuiFlexItem grow={false}>
            {typeof props.description === 'string' ? (
              <EuiText>
                <p>{props.description}</p>
              </EuiText>
            ) : (
              <React.Fragment>{props.description}</React.Fragment>
            )}
          </EuiFlexItem>
        </EuiFlexGroup>
        <EuiSpacer size="s" />
      </EuiModalBody>

      <EuiModalFooter>
        <EuiButtonEmpty data-test-subj="cancelButton" onClick={props.onCancel}>
          Cancel
        </EuiButtonEmpty>

        <EuiButton
          data-test-subj="confirmButton"
          color={props.confirmButtonColor}
          fill
          onClick={props.onConfirm}
          disabled={!!props.confirmButtonDisabled}
        >
          {props.confirmButtonText}
        </EuiButton>
      </EuiModalFooter>
    </EuiModal>
  );
};
