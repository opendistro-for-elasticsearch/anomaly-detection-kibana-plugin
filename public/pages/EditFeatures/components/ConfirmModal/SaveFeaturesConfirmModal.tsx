/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiFlexItem,
  EuiFlexGroup,
  EuiText,
  EuiRadioGroup,
} from '@elastic/eui';
import React from 'react';
// @ts-ignore
import { toastNotifications } from 'ui/notify';
//@ts-ignore
import chrome from 'ui/chrome';
import { ConfirmModal } from '../../../DetectorDetail/components/ConfirmModal/ConfirmModal';
import { SAVE_FEATURE_OPTIONS } from '../../utils/constants';

interface SaveFeaturesConfirmModalProps {
  readyToStartAdJob: boolean;
  saveFeatureOption: SAVE_FEATURE_OPTIONS;
  onClose(): void;
  onCancel(): void;
  onConfirm(): void;
  onOptionChange(id: string): void;
}

export function SaveFeaturesConfirmModal(props: SaveFeaturesConfirmModalProps) {
  const startAdJobOptions = (disableStartAdJob: boolean) => {
    return [
      {
        id: SAVE_FEATURE_OPTIONS.START_AD_JOB,
        label: 'Automatically run detector (Recommended)',
        disabled: disableStartAdJob,
      },
      {
        id: SAVE_FEATURE_OPTIONS.KEEP_AD_JOB_STOPPED,
        label: 'Manually start the detector at a later time',
      },
    ];
  };

  const confirmModalDescription = () => (
    <EuiFlexGroup direction="column">
      <EuiFlexItem grow={false}>
        <EuiText>
          <p>
            The detector is currently stopped. To receive accurate and real-time
            anomalies, the detector need to start and collect sufficient data to
            include your latest change. The earlier the detector starts running,
            the sooner the anomalies will be available.
          </p>
        </EuiText>
      </EuiFlexItem>
      <EuiFlexItem grow={false}>
        <EuiRadioGroup
          name="start ad radio group"
          options={startAdJobOptions(!props.readyToStartAdJob)}
          idSelected={props.saveFeatureOption}
          onChange={props.onOptionChange}
        />
      </EuiFlexItem>
    </EuiFlexGroup>
  );

  return (
    <ConfirmModal
      title="Automatically start the detector?"
      description={confirmModalDescription()}
      confirmButtonText="Confirm"
      confirmButtonColor="primary"
      onClose={props.onClose}
      onCancel={props.onCancel}
      onConfirm={props.onConfirm}
    />
  );
}
