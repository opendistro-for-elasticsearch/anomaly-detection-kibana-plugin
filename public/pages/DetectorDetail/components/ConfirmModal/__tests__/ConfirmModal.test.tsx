/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ConfirmModal } from '../ConfirmModal';

describe('<ConfirmModal /> spec', () => {
  const onClose = jest.fn();
  const onCancel = jest.fn();
  const onConfirm = jest.fn();
  const component = <div>mock component</div>;
  describe('Confirm Modal', () => {
    test('renders component with callout', () => {
      const { container } = render(
        <ConfirmModal
          title="test confirm modal"
          description="this is a testing description"
          callout="test callout"
          confirmButtonText=""
          confirmButtonColor="primary"
          onClose={onClose}
          onCancel={onCancel}
          onConfirm={onConfirm}
        />
      );
      expect(container).toMatchSnapshot();
    });

    test('renders component with empty callout', () => {
      const { container } = render(
        <ConfirmModal
          title="test confirm modal"
          description="this is a testing description"
          confirmButtonText=""
          confirmButtonColor="primary"
          onClose={onClose}
          onCancel={onCancel}
          onConfirm={onConfirm}
        />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
