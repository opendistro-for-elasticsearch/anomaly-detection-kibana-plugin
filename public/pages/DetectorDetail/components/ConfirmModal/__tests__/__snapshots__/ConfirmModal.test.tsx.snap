// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmModal /> spec Confirm Modal renders component with callout 1`] = `
<div>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="1"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="euiModal euiModal--maxWidth-default"
      tabindex="0"
    >
      <button
        aria-label="Closes this modal window"
        class="euiButtonIcon euiButtonIcon--text euiModal__closeIcon"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
      </button>
      <div
        class="euiModal__flex"
      >
        <div
          class="euiModalHeader"
        >
          <div
            class="euiModalHeader__title"
          >
            test confirm modal
          </div>
        </div>
        <div
          class="euiModalBody"
          style="padding-left: 24px; padding-right: 24px;"
        >
          <div
            class="euiModalBody__overflow"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                test callout
              </div>
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  class="euiText euiText--medium"
                >
                  <p>
                    this is a testing description
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiModalFooter"
        >
          <button
            class="euiButtonEmpty euiButtonEmpty--primary"
            type="button"
          >
            <span
              class="euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                Cancel
              </span>
            </span>
          </button>
          <button
            class="euiButton euiButton--primary euiButton--fill"
            type="button"
          >
            <span
              class="euiButton__content"
            >
              <span
                class="euiButton__text"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;

exports[`<ConfirmModal /> spec Confirm Modal renders component with empty callout 1`] = `
<div>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="1"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="euiModal euiModal--maxWidth-default"
      tabindex="0"
    >
      <button
        aria-label="Closes this modal window"
        class="euiButtonIcon euiButtonIcon--text euiModal__closeIcon"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
      </button>
      <div
        class="euiModal__flex"
      >
        <div
          class="euiModalHeader"
        >
          <div
            class="euiModalHeader__title"
          >
            test confirm modal
          </div>
        </div>
        <div
          class="euiModalBody"
          style="padding-left: 24px; padding-right: 24px;"
        >
          <div
            class="euiModalBody__overflow"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  class="euiText euiText--medium"
                >
                  <p>
                    this is a testing description
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiModalFooter"
        >
          <button
            class="euiButtonEmpty euiButtonEmpty--primary"
            type="button"
          >
            <span
              class="euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                Cancel
              </span>
            </span>
          </button>
          <button
            class="euiButton euiButton--primary euiButton--fill"
            type="button"
          >
            <span
              class="euiButton__content"
            >
              <span
                class="euiButton__text"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;
